andela/Ah-backend-groot
# #164069327 Highlight and comment on any text in an article

**What does this PR do?**
It ensures a user can highlight and comment on any text in an article.

**Background context**
A comment can only be made on a particular article and to highlight a particular section of the article, it has to be as long as the indexing or else nothing will be highlighted. The `start_position` and `end_position` represent the position of each letter in the article.

**How should this manually be tested**
- Login and retrieve an access token
- Use this token to create articles
- After creating articles, post a comment about that article at this endpoint:
`articles/<str:slug>/comments/`
- The comment should be in this format, specifying the area of the article you wish to slice and comment on.
`"{
      "comment":{
                             "body":"body goes here.",
                             "start_position":<int>,
                             "end_position":<int>
   }
}

**Screenshots**
The highlighted blue area shows the article section that has been sliced for highlighting.
![image](https://user-images.githubusercontent.com/28726900/54573603-05d6e100-49fe-11e9-9d51-1b461bb73e9e.png)
**What are the relevant PT stories?**
#164069327
